doc/releases/changelog-dev.md

```diff
@@ -39,6 +39,37 @@
   is now strictly faster for jax.
   [(#4963)](https://github.com/PennyLaneAI/pennylane/pull/4963)
 
+* New `qml.commutator` function that allows to compute commutators between
+  `qml.operation.Operator`, `qml.pauli.PauliWord` and `qml.pauli.PauliSentence` instances.
+  [(#5051)](https://github.com/PennyLaneAI/pennylane/pull/5051)
+
+  Basic usage with PennyLane operators.
+
+  ```pycon
+  >>> qml.commutator(qml.PauliX(0), qml.PauliY(0))
+  2j*(PauliZ(wires=[0]))
+  ```
+
+  We can return a `PauliSentence` instance by setting `pauli=True`.
```

Comment:
We can discuss a bit more offline, but I still feel like the `pauli=True` is a bit of an implementation detail that might not need to be exposed to the user.

Reply:
I took inspiration from the documentation of `qml.dot` here
![image](https://github.com/PennyLaneAI/pennylane/assets/43949391/913fce7a-e098-4291-a328-1d16f1aeab79)

Do you mean just not mentioning the possibility of `pauli=True` in the docs or entirely removing it from the function?

Reply:
I think long term the goal should be (as you suggested) to do these speedups under the hood by checking if the operator has a pauli representation

Reply:
@trbromley I opened a follow-up PR doing exactly that. There are, however, some concerns/potential blockers as outlined in the PR description, which is why I would decouple that from this PR here.

https://github.com/PennyLaneAI/pennylane/pull/5098

Reply:
Thanks! Will take a look.